src/plugins/dist_matrixops/dist_inverse_operation.cpp

```diff
@@ -0,0 +1,491 @@
+
+                            // Find the locality that owns the pivot element
+                            // then get the pivot
+                            std::size_t ownid2 = findOwningLoc(
+                                numRows, numLocalities, current_row);
+                            std::size_t localIndexOffset2 = current_row -
+                                getStartCol(ownid2, numRows, numLocalities);
+                            auto pulledElement =
+                                lhs_data
+                                    .fetch(ownid2, nextRow, localIndexOffset2,
+                                         nextRow + 1,
+                                         localIndexOffset2 + 1)
+                                    .get();
+                            double factor = pulledElement(0, 0);
+
+                            // Removing this barrier causes errors
+                            if (numLocalities > 1)
+                            {
+                                hpx::lcos::barrier b5("barriere_" +
+                                        lhs_localities.annotation_.name_,
+                                    lhs_localities.locality_.num_localities_,
+                                    lhs_localities.locality_.locality_id_);
+                                b5.wait();
+                            }
+
+
+                            for (std::size_t nextCol = 0;
+                                 nextCol != numCols; nextCol++)
+                            {
+                                (*lhs_data)(nextRow, nextCol) =
+                                    (*lhs_data)(nextRow, nextCol) -
+                                    (factor *
+                                        (*lhs_data)(current_row, nextCol));
+                                invMatrix(nextRow, nextCol) =
+                                    invMatrix(nextRow, nextCol) -
+                                    (factor * invMatrix(current_row, nextCol));
+                            }
+                            //if (numLocalities > 1)
+                            //{
+                            //    hpx::lcos::barrier b6("barrierf" +
+                            //            std::to_string(nextRow) +
+                            //            "_" +
+                            //            lhs_localities.annotation_.name_,
+                            //        lhs_localities.locality_.num_localities_,
+                            //        lhs_localities.locality_.locality_id_);
+                            //    b6.wait();
+                            //}
+
+                        }
+                    }
+                }
+
+               //hpx::lcos::barrier b7("barrierg" + std::to_string(current_row) +
+               //         "_" + lhs_localities.annotation_.name_,
+               //     lhs_localities.locality_.num_localities_,
+               //     lhs_localities.locality_.locality_id_);
+               // b7.wait();
```

Comment:
Would you mind removing the commented code in this file?

Reply:
I only the removed the comments that were notes to myself (such as "this is mostly boilerplate, just change 'X,Y,Z' ") and the commented out barriers. The comments that describe the functionality are still there, but I can definitely add back the other comments if that would be better.